src/Microsoft.DotNet.Build.Tasks/GenerateCurrentVersion.cs

```diff
@@ -33,12 +39,29 @@ public sealed class GenerateCurrentVersion : Task
         [Output]
         public string GeneratedVersion { get; set; }
 
+        /// <summary>
+        /// The Revision number that will be produced from the BuildNumber.
+        /// </summary>
+        [Output]
+        public string GeneratedRevision { get; set; }
+
         private const string DateFormat = "yyyy-MM-dd";
         private const string LastModifiedTimeDateFormat = "yyyy-MM-dd HH:mm:ss.FFFFFFF";
         private CultureInfo enUS = new CultureInfo("en-US");
 
         public override bool Execute()
         {
+            // Calculating GeneratedRevision
+            if (string.IsNullOrEmpty(OfficialBuildId))
+            {
+                GeneratedRevision = "00";
```

Comment:
We should also update this to be "0" and our targets file IIRC you had an instance of that.